src/components/PopupWithForm.js

```diff
@@ -1,19 +1,33 @@
 import Popup from "./Popup.js";
 
 export default class PopupWithForm extends Popup {
-  constructor(popupSelector, handleFormSubmit) {
+  constructor(popupSelector, handleFormSubmit, initialButtonText) {
     super({ popupSelector });
     this._popupForm = this._popupElement.querySelector(".modal__form");
     console.log(this._popupForm);
     this._handleFormSubmit = handleFormSubmit;
     console.log(this._closeButton);
+    this._submitButton = this._popupForm.querySelector(".modal__button");
+    this._initialButtonText = initialButtonText;
   }
 
   close() {
     this._popupForm.reset();
     super.close();
   }
 
+  renderLoading(isLoading) {
+    if (isLoading) {
+      if (this._initialButtonText === "Save") {
+        this._submitButton.textContent = "Saving...";
+      } else if (this._initialButtonText === "Create") {
+        this._submitButton.textContent = "Creating...";
+      }
+    } else {
+      this._submitButton.textContent = this._initialButtonText;
+    }
+  }
+
```

Comment:
**Could be improved**

You can make ` setInputValues` in class `PopupWithForm`, which can insert data into inputs:

```js

  setInputValues(data) {
    this._inputList.forEach((input) => {
      // here you insert the `value` by the `name` of the input
      input.value = data[input.name];
    });
  }



```

So, this way you’ll not have to search the inputs of the profile in `index.js`